cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/data/CassandraDataLayer.java

```diff
@@ -685,6 +700,7 @@ private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundE
         }
         this.rfMap = (Map<String, ReplicationFactor>) in.readObject();
         this.initInstanceMap();
+        this.startupValidate();
```

Comment:
Shouldn't this be performed _before_ `initInstanceMap`? `initInstanceMap` will likely reach out to external services to build the instance map and may fail to retrieve information? This would defeat the purpose of using `startupValidate`?

Reply:
initInstanceMap, will instantiate the sidecar client, which is needed to perform the connection checks